libs/core/src/lib/popover/cdk-overlay/cdk-popover.component.ts

```diff
@@ -0,0 +1,395 @@
+                this.isOpenChange.emit(false);
+            }
+            this.isOpen = false;
+        }
+    }
+
+    /** Opens the popover. */
+    open(): void {
+        if (!this._overlayRef || !this._overlayRef.hasAttached()) {
+            this._overlayRef = this._overlay.create(this._getOverlayConfig());
+            this._overlayRef.attach(new TemplatePortal(this.templateRef, this.container));
+
+            this._changeDetectorReference.detectChanges();
+
+            if (!this.isOpen) {
+                this.isOpenChange.emit(true);
+            }
+            this.isOpen = true;
+
+            if (this.fillControlMode) {
+                this._listenOnResize();
+            }
+
+            this._listenOnClose();
+            this._listenOnOutClicks();
+        }
+    }
+
+    /**
+     * Method called to change position of popover,
+     * recommended to be used only when popover is opened, otherwise change position or cdkPlacement
+     */
+    applyNewPosition(positions: ConnectedPosition[]): void {
+        const refPosition = this._getPositionStrategy(positions);
+        this._listenForPositionChange(refPosition.positionChanges);
+        this._overlayRef.updatePositionStrategy(refPosition);
+    }
+
+    /** Method called to refresh position of opened popover */
+    refreshPosition(): void {
+        if (this._overlayRef) {
+            this._overlayRef.updatePosition();
+        }
+    }
+
+    /** Handler for alt + arrow down keydown */
+    triggerKeyDownHandler(event: KeyboardEvent): void {
+        if (KeyUtil.isKey(event, 'ArrowDown') && event.altKey && !this.isOpen) {
+            this.open();
+        }
+    }
+
+    /** Handler escape keydown */
+    bodyKeydownHandler(event: KeyboardEvent): void {
+        if (KeyUtil.isKey(event, 'Escape') && this.isOpen && this.closeOnEscapeKey) {
+            this.close();
+        }
+    }
+
+    private get _triggerElement(): ElementRef<any> {
+        return this.trigger || this.triggerOrigin.elementRef;
+    }
+
+    /** Subscribe to close events from CDK Overlay, to throw proper events, change values */
+    private _listenOnClose(): void {
+        this._overlayRef.detachments()
+            .pipe(takeUntil(this._refresh$))
+            .subscribe(() => this.close())
+        ;
+    }
+
+    /** Listener for click events */
+    private _listenOnOutClicks(): void {
+        this._overlayRef.backdropClick().pipe(
+            filter(event => this._shouldClose(event)),
+            takeUntil(this._refresh$)
+        ).subscribe(() => this.close());
+
+        this._overlayRef._outsidePointerEvents.pipe(
+            filter(event => this._shouldClose(event)),
+            takeUntil(this._refresh$)
+        ).subscribe(() => this.close());
+    }
+
+    /** Refresh listeners on trigger element events */
+    private _refreshTriggerListeners(): void {
+        this._removeTriggerListeners();
+        if (this.triggers && this.triggers.length > 0) {
```

Comment:
Can be simplified to `if (this.triggers?.length > 0)` or even `if (this.triggers?.length)`